python/lsst/daf/butler/script/exportCalibs.py

```diff
@@ -69,23 +70,23 @@ def parseCalibrationCollection(
     RuntimeError
         Raised if the collection to search is not a CALIBRATION collection.
     """
-    if registry.getCollectionType(collection) != CollectionType.CALIBRATION:
+    if butler.registry.getCollectionType(collection) != CollectionType.CALIBRATION:
         raise RuntimeError(f"Collection {collection} is not a CALIBRATION collection.")
 
-    exportCollections = []
+    exportCollections = set()
     exportDatasets = []
     for calibType in datasetTypes:
-        associations = registry.queryDatasetAssociations(
-            calibType, collections=collection, collectionTypes=[CollectionType.CALIBRATION]
-        )
-        for result in associations:
-            # Need an expanded dataId in case file templates will be used
-            # in the transfer.
-            dataId = registry.expandDataId(result.ref.dataId)
-            ref = result.ref.expanded(dataId)
-            exportDatasets.append(ref)
-            assert ref.run is not None, "These refs must all be resolved."
-            exportCollections.append(ref.run)
+        with butler._query() as query:
+            results = query.datasets(calibType, collections=collection, find_first=False)
+
+            try:
+                refs = list(results.with_dimension_records())
+            except Exception as e:
+                e.add_note(f"Error from querying dataset type {calibType} and collection {collection}")
+                raise
+            exportDatasets.extend(refs)
+            exportCollections.update([ref.run for ref in refs])
```

Comment:
I don't think this is necessary; `RUN` collections are always and only exported implicitly when datasets from them are exported.